File: Wisej.Web.Ext.Ignite/Wisej.Web.Ext.Ignite/igPieChart.cs
///////////////////////////////////////////////////////////////////////////////
//
// (C) 2020 ICE TEA GROUP LLC - ALL RIGHTS RESERVED
//
// 
//
// ALL INFORMATION CONTAINED HEREIN IS, AND REMAINS
// THE PROPERTY OF ICE TEA GROUP LLC AND ITS SUPPLIERS, IF ANY.
// THE INTELLECTUAL PROPERTY AND TECHNICAL CONCEPTS CONTAINED
// HEREIN ARE PROPRIETARY TO ICE TEA GROUP LLC AND ITS SUPPLIERS
// AND MAY BE COVERED BY U.S. AND FOREIGN PATENTS, PATENT IN PROCESS, AND
// ARE PROTECTED BY TRADE SECRET OR COPYRIGHT LAW.
//
// DISSEMINATION OF THIS INFORMATION OR REPRODUCTION OF THIS MATERIAL
// IS STRICTLY FORBIDDEN UNLESS PRIOR WRITTEN PERMISSION IS OBTAINED
// FROM ICE TEA GROUP LLC.
//
///////////////////////////////////////////////////////////////////////////////


namespace Wisej.Web.Ext.Ignite
{
	/// <summary>
	/// The igPieChart is a charting control for rendering pie charts in HTML5 Web applications and sites.
	/// </summary>
	public class igPieChart : igBase
	{

		#region Constructors

		/// <summary>
		/// Constructs a new instance of the <see cref="igPieChart"/> class.
		/// </summary>
		public igPieChart() : this(null)
		{
		}

		/// <summary>
		/// Constructs a new instance of the <see cref="igPieChart"/> class.
		/// </summary>
		/// <param name="options"></param>
		public igPieChart(object options) : base("igPieChart", options)
		{
			this.WidgetWiredEvents = new[] {
			"browserNotSupported",
			"labelClick",
			"selectedItemChanged",
			"selectedItemChanging",
			"selectedItemsChanged",
			"selectedItemsChanging",
			"sliceClick",
			"tooltipHidden",
			"tooltipHiding",
			"tooltipShowing",
			"tooltipShown"
			};
		}

		#endregion
	}
}

File: Wisej.Web.Ext.Ignite/Wisej.Web.Ext.Ignite/igPieChart.js
﻿//# sourceURL=wisej.web.ext.IgniteWidget.igPieChart.js

///////////////////////////////////////////////////////////////////////////////
//
// (C) 2020 ICE TEA GROUP LLC - ALL RIGHTS RESERVED
//
// 
//
// ALL INFORMATION CONTAINED HEREIN IS, AND REMAINS
// THE PROPERTY OF ICE TEA GROUP LLC AND ITS SUPPLIERS, IF ANY.
// THE INTELLECTUAL PROPERTY AND TECHNICAL CONCEPTS CONTAINED
// HEREIN ARE PROPRIETARY TO ICE TEA GROUP LLC AND ITS SUPPLIERS
// AND MAY BE COVERED BY U.S. AND FOREIGN PATENTS, PATENT IN PROCESS, AND
// ARE PROTECTED BY TRADE SECRET OR COPYRIGHT LAW.
//
// DISSEMINATION OF THIS INFORMATION OR REPRODUCTION OF THIS MATERIAL
// IS STRICTLY FORBIDDEN UNLESS PRIOR WRITTEN PERMISSION IS OBTAINED
// FROM ICE TEA GROUP LLC.
//
///////////////////////////////////////////////////////////////////////////////

// Returns a data map that can be converted to JSON.
this.filterEventData = function (args) {

    switch (args.type) {

        case "sliceClick":
            if (args.slice.item.__inner)
                return args.slice.item.__inner;
            else
                return args.slice.item;
            break;

        case "selectedItemChanged":
        case "selectedItemChanging":
            return {
                oldItem: args.oldItem,
                newItem: args.newItem
            };
            break;

        case "selectedItemsChanged":
        case "selectedItemsChanging":
            return {
                oldItems: args.oldItems,
                newItems: args.newItems
            };
            break;

        case "labelClick":
            return args.item;
            break;

        default:
            return args;
            break;

    }
};
